File: ruby/stack/test_stack.rb
require_relative 'stack'
require 'test/unit'

class TestLinkedList < Test::Unit::TestCase
  def setup
    @e1 = Element.new(1)
    @e2 = Element.new(2)
    @e3 = Element.new(3)
    @e4 = Element.new(4)
    @stack = Stack.new(@e1)
  end

  def test_stack
    @stack.push(@e2)
    @stack.push(@e3)
    assert_equal(@stack.pop.value, 3)
    assert_equal(@stack.pop.value, 2)
    assert_equal(@stack.pop.value, 1)
    assert_nil(@stack.pop)
    @stack.push(@e4)
    assert_equal(@stack.pop.value, 4)
  end
end

File: python/test_stack.py
import unittest
from stack import *

class StackTest(unittest.TestCase):
    def setUp(self):
        self.e1 = Element(1)
        self.e2 = Element(2)
        self.e3 = Element(3)
        self.e4 = Element(4)
        self.stack = Stack(self.e1)


    def test_stack(self):
        self.stack.push(self.e2)
        self.stack.push(self.e3)
        self.assertEqual(self.stack.pop().value, 3)
        self.assertEqual(self.stack.pop().value, 2)
        self.assertEqual(self.stack.pop().value, 1)
        self.assertIsNone(self.stack.pop())
        self.stack.push(self.e4)
        self.assertEqual(self.stack.pop().value, 4)

if __name__ == '__main__':
    unittest.main()
